reader.go

```diff
@@ -93,12 +116,83 @@ func (r *Reader) NumRead() int {
 
 // LastModified returns the last modified time of the remote feed.
 func (r *Reader) LastModified() (time.Time, error) {
-	lastMod, err := remoteLastModified(r.ctx, r.remote)
-	return lastMod.Time(), err
+	var lastMod timestamp
+	for _, remote := range r.remotes {
+		t, err := remoteLastModified(r.ctx, remote)
+		if err != nil {
+			return time.Time{}, err
+		}
+		if t > lastMod {
+			lastMod = t
+		}
+	}
+
+	return lastMod.Time(), nil
 }
 
 // Close closes the reader.
 func (r *Reader) Close() error {
+	if r.cur != nil {
+		return r.cur.Close()
+	}
+	return nil
+}
+
+func (r *Reader) ensureCurrent() bool {
+	if r.pos >= len(r.remotes) {
+		return false
+	}
+
+	if r.cur == nil {
+		remote := r.remotes[r.pos]
+		o := r.opt
+		o.norm(remote.Name())
```

Comment:
I don't think this will work. the first time you call `norm`, you will set some values which means they will not be overwritten the next time you call `norm`. although.. you are copying `r.opt` .... I think it would be better to keep the passed `opt` pointer attached to the struct and init new opt here, for example:

```go
var opt ReaderOptions
if r.opt != nil {
  opt = *r.opt
}
opt.norm(remote.Name())
```

I think your code is fine, but it's always better to be defensive (and more readable)